engine/verification/requester/requester.go

```diff
@@ -191,16 +191,25 @@ func (e *Engine) handleChunkDataPack(originID flow.Identifier, chunkDataPack *fl
 	e.metrics.OnChunkDataPackResponseReceivedFromNetworkByRequester()
 
 	// makes sure we still need this chunk, and we will not process duplicate chunk data packs.
-	removed := e.pendingRequests.Rem(chunkID)
+	locators, removed := e.pendingRequests.PopAll(chunkID)
```

Comment:
> Suppose EN1 and EN2 produce 2 execution results for the same block, which have different IDs and constitute an execution fork. We are processing chunk C.

Let's formulate it more: suppose we have `result1` and `result2` both for the same block, but with different result IDs, hence an execution fork. For each chunk `c1` in `result1` and `c2` in `result2` that _share the same chunk index_ there will be either of the following cases:

- Case 1: `c1` and `c2` are two _distinct_ chunks. Then they have distinct chunk IDs. Hence each of these chunks are verified separately, i.e., a separate chunk data request for `c1` and a separate one for `c2` and then they take their exclusive verification path. This is the existing verification logic on `master` and this PR does not change it. 

- Case 2: `c1` and `c2` are _identical_ chunks. Then they have the same chunk ID. Although each of these chunks again are treated distinctly and have their own verification path, they share their chunk data pack. This follows the assumption that chunk data packs are matched with chunk IDs. Hence, only a single chunk data pack is requested for both of these chunks. Once that chunk data pack arrives, both of these chunks `c1` and `c2` resume their verification path separately. This was not previously supported, and is the core functionality added by this PR. Without this PR in place, if a verification node is assigned both `c1` and `c2` concurrently, it deduplicates one of them in the favor of the other one. However, since each of them belong to a distinct result, and since result approvals are attributed to the execution results, one result would end up having no approval for its chunk. This situation happend on testnet and the PR is fixing that. 

Note that in both cases `c1` and `c2` will have their own result approvals.